### PR TITLE
SY-1611 Bring Back Line Plot Annotation Tool

### PR DESCRIPTION
**Pull request by pjdotson**
# Issue Pull Request

## Key Information

- **Linear Issue**: [SY-1611](https://linear.app/synnax/issue/SY-1611/bring-back-annotation-tool)

## Description

Added the rules annotations back to the line plot visualization. Refactored some code to improve readability. Add a median to the `x/ts/bounds`. Couple QOL improvements to annotations (scrolling, better preset rules).

## Basic Readiness

- [x] I have performed a self-review of my code.
- [x] I have added relevant tests to cover the changes to CI.
- [x] I have added needed QA steps to the [release candidate](/synnaxlabs/synnax/blob/main/.github/PULL_REQUEST_TEMPLATE/rc.md) template that cover these changes.
- [x] I have updated in-code documentation to reflect the changes.
- [x] I have updated user-facing documentation to reflect the changes.

### Data Structures

I have ensured that previous versions of stored data structures are properly migrated to new formats in the following projects:

- [x] Server
- [x] Console

### API Changes

The following projects have backwards-compatible APIs:

- [x] Python Client
- [x] Server
- [x] TypeScript Client

### CODE REVIEW

File: x/ts/src/spatial/bounds/bounds.ts [hunk 1]
```diff
@@ -191,6 +191,22 @@ export const isFinite = (a: Crude): boolean => {
...
+ *
+ * @param a - The bounds to find the median of. Can be either a strict bounds object
+ * with 'lower' and 'upper' properties or an array of length 2.
+ * @returns The median value between the lower and upper bounds.
+ *
+ * @example
+ * bounds.median([0, 10]) // => 5
+ * bounds.median({ lower: 0, upper: 10 }) // => 5
+ */
+export const median = (a: Crude): number => {
```
Comment (emilbon99) [inline x/ts/src/spatial/bounds/bounds.ts:205]: This seems more like a mean than a median

**Review by emilbon99 — CHANGES_REQUESTED** [review]
Screenshot 2024-12-02 at 3 57 17 PM

Double clicking to edit the label on the rule does not work

**Comment by emilbon99** [comment]
Screenshot 2024-12-06 at 10 43 14 AM

**Comment by pjdotson** [comment]
> Screenshot 2024-12-06 at 10 43 14 AM

@emilbon99 I can't get the issues with the reseting position and rounding to replicate.

**Pull request merged**